cmd/oras/root/manifest/fetch.go

```diff
@@ -130,14 +149,37 @@ func fetchManifest(cmd *cobra.Command, opts *fetchOptions) (fetchErr error) {
 			return fmt.Errorf("failed to fetch the content of %q: %w", opts.RawReference, err)
 		}
 
-		if opts.outputPath == "" || opts.outputPath == "-" {
-			// output manifest content
-			return opts.Output(os.Stdout, content)
+		if opts.Template != "" {
+			if opts.Template == "json" {
+				// output prettified json manifest content
+				opts.Pretty.Pretty = true
+				if err := opts.Output(os.Stdout, content); err != nil {
+					return err
+				}
+			} else {
+				// output formatted data
+				switch desc.MediaType {
+				case ocispec.MediaTypeImageManifest, docker.MediaTypeManifest:
+					var manifest ocispec.Manifest
+					if err := json.Unmarshal(content, &manifest); err != nil {
+						return err
+					}
+					if err = handler.OnFetched(manifest); err != nil {
+						return err
+					}
+				default:
+					return fmt.Errorf("cannot apply template to %q: unsupported media type %s", opts.RawReference, desc.MediaType)
+				}
+			}
 		}
```

Comment:
This part does not seem correct. They should be refactored to "metadata" output for different types.